packages/core/core/src/services/entity-validator/validators.ts

```diff
@@ -380,8 +377,7 @@ const addUniqueValidator = <T extends yup.AnySchema>(
     }
 
     // The validation should pass if there is no other record found from the query
-    const queryUid = model.uid;
-    return !(await strapi.db.query(queryUid).findOne({ where: scalarAttributeWhere }));
+    return !(await strapi.db.query(model.uid).findOne({ where: scalarAttributeWhere }));
```

Comment:
It probably doesn't make much of a difference, but maybe selecting only the id field could make the query faster?

Reply:
sure!